plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkTask.kt

```diff
@@ -4,132 +4,257 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
-import org.gradle.api.tasks.InputFile
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.*
 import org.gradle.internal.jvm.Jvm
```

Comment:
❌  Internal API usage.

Reply:
What alternatives do you propose?

Reply:
What's the use case or what do you need?

Reply:
To get the build JDK to determine the location of the `jlink` and `jdeps` tools. The `java.home` system property can't be used since Gradle may be running on a different JDK, like if `-Dorg.gradle.java.home` is set

Reply:
Can you open an issue with Gradle or ask about an alternative solution in the Gradle Slack channel?